File: lms-frontend/src/employee/services/shiftService.js
// shiftService.js

import api from "../../api/axios";

const BASE_URL = "/api/shifts";

const getAllShifts = () => {

    return api.get(
        BASE_URL
    );
};

const getShiftById = (shiftId) => {

    return api.get(
        `${BASE_URL}/${shiftId}`
    );
};

const createShift = (shiftData) => {

    return api.post(
        BASE_URL,
        shiftData
    );
};

const updateShift = (
    shiftId,
    shiftData
) => {

    return api.put(
        `${BASE_URL}/${shiftId}`,
        shiftData
    );
};

const deleteShift = (shiftId) => {

    return api.delete(
        `${BASE_URL}/${shiftId}`
    );
};

const shiftService = {

    getAllShifts,
    getShiftById,
    createShift,
    updateShift,
    deleteShift
};

export default shiftService;

File: backend/src/main/java/com/lms_erp/employee/service/ShiftService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.ShiftRequest;
import com.lms_erp.employee.dto.ShiftResponse;

import java.util.List;

public interface ShiftService {

    ShiftResponse createShift(ShiftRequest request);

    ShiftResponse updateShift(Long shiftId,
                              ShiftRequest request);

    ShiftResponse getShiftById(Long shiftId);

    List<ShiftResponse> getAllShifts();

    void deleteShift(Long shiftId);
}